### ISSUE TITLE
RSS Feed shows unapproved comments (if moderate=1)

### ISSUE THREAD

**Issue by xrat** [dokufreaks/plugin-discussion#193]
If `$conf['plugin']['discussion']['moderate'] = 1;` (moderate new comments), an RSS feed generated by the feed plugin (for example) shows comments which haven't been approved yet. This is a rather critical issue for public DokuWiki installations as spam comments will show up in the feed.

This issue might very well affect more than what I found. And all I can provide is a quick fix for the feeds: In `helper.php`, in `function _handleRecentComment()` add the following line
```
if ($data['comments'][$cid]['show'] != 1) return false;
```
after e.g.
```
// check if the comment still exists
if (!isset($data['comments'][$cid])) return false;
```

Affected: At least discussion plugin v2015-05-17 installed from https://github.com/dokufreaks/plugin-discussion/archive/master.zip

**Comment by michitux** [comment]
Good observation, thank you!

I think that we actually also need to check if all parents exist and are visible due to how visibility is decided in the discussion plugin. I have pushed an attempt at fixing this in the pull request #197. Could you (or somebody else) please verify if this works? Thank you!

**Comment by xrat** [comment]
@michitux Thanks a lot for the fix! I have only 1 [installation](https://fam.tuwien.ac.at/~schamane/_/) where I could easily test #197 but better this than nothing. Result: All OK!
Details: I reverted my hack, added a comment in my [blog namespace](https://fam.tuwien.ac.at/~schamane/_/blog:141118_suppress_ip_of_authenticated_senders_in_sendmail) that was not approved, i.e. "hidden", and I checked that it showed up in the feed. Then I added my hack back in, removed the feed's cache, and the comment did not show up. Then replaced `helper.php` with the version from #197, removed the cache file again, and the comment still did not show up. I approved the comment ("show"), removed the cache and it showed up. I am not sure how to test all parents, though.
